File: C++/2.5.2.cpp
#include <iostream>
#include <iomanip>
#include <vector>

using namespace std;

int main() {
    const int MAX_SIZE = 100;
    int n;
    double sumDiagonal = 0.0;

    // Вводим размер матрицы
    cout << "Enter the size of the matrix n: ";
    cin >> n;

    // Проверка на корректность размера матрицы
    if (n <= 0 || n > MAX_SIZE) {
        cout << "The size of the matrix must be between 1 and " << MAX_SIZE << endl;
        return 1;
    }

    // Инициализируем матрицу
    vector<vector<double>> A(n, vector<double>(n));

    // Вводим данные для матрицы
    cout << "Enter the elements of the matrix:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << "Enter element A[" << i + 1 << "," << j + 1 << "]: ";
            cin >> A[i][j];
        }
    }

    // Выводим матрицу после ввода данных
    cout << "\nMatrix after input:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << fixed << setprecision(2) << A[i][j] << " ";
        }
        cout << endl;
    }

    // Двукратный цикл для изменения элементов под главной диагональю и вычисления суммы главной диагонали
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == j) {
                sumDiagonal += A[i][j];
            } else if (i > j) {
                if (A[i][j] > A[j][i]) {
                    A[i][j] = (A[i][j] + A[j][i]) / 2.0;
                }
            }
        }
    }

    // Выводим измененную матрицу
    cout << "\nModified matrix:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << fixed << setprecision(2) << A[i][j] << " ";
        }
        cout << endl;
    }

    // Выводим сумму элементов главной диагонали
    cout << "Sum of the elements of the main diagonal: " << fixed << setprecision(2) << sumDiagonal << endl;

    return 0;
}

File: pascal/2.5.2.pas
const
  MAX_SIZE = 100;

var
  n, i, j: Integer;
  A: array[1..MAX_SIZE, 1..MAX_SIZE] of Real;
  sumDiagonal: Real;

begin
  // Вводим размер матрицы
  Write('Enter the matrix size n: ');
  ReadLn(n);

  if n > MAX_SIZE then
    n := MAX_SIZE;

  // Вводим данные для матрицы
  for i := 1 to n do
    for j := 1 to n do
    begin
      Write('Enter element A[', i, ',', j, ']: ');
      ReadLn(A[i, j]);
    end;

  // Инициализируем сумму элементов главной диагонали
  sumDiagonal := 0;

  // Двукратный цикл для изменения элементов под главной диагональю и вычисления суммы главной диагонали
  for i := 1 to n do
  begin
    for j := 1 to n do
    begin
      if i = j then
        sumDiagonal := sumDiagonal + A[i, j]
      else if i > j then
      begin
        if A[i, j] > A[j, i] then
          A[i, j] := (A[i, j] + A[j, i]) / 2;
      end;
    end;
  end;

  // Выводим измененную матрицу
  WriteLn('Modified matrix:');
  for i := 1 to n do
  begin
    for j := 1 to n do
      Write(A[i, j]:0:2, ' ');
    WriteLn;
  end;

  // Выводим сумму элементов главной диагонали
  WriteLn('Sum of the elements of the main diagonal: ', sumDiagonal:0:2);
end.